widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.kt

```diff
@@ -1262,13 +1262,17 @@ class ChatController(
     ): List<ChatMessageInternal>? {
         return if (newHistory.isNullOrEmpty() || oldHistory.isNullOrEmpty()) {
             newHistory
-        } else newHistory.filter { isNewMessage(it.chatMessage) }
+        } else newHistory.filter { isNewMessage(oldHistory, it.chatMessage) }
     }
 
-    private val oldMessageIdsHash: MutableSet<String> by lazy { mutableSetOf() }
-
-    private fun isNewMessage(newMessage: ChatMessage): Boolean =
-        oldMessageIdsHash.add(newMessage.id)
+    private fun isNewMessage(oldHistory: List<ChatItem>?, newMessage: ChatMessage): Boolean =
+        oldHistory
```

Comment:
I would do the following:
* start with a let to get rid of the nullable, if oldHistory is null, then return true
* remove the type declarations, they are not needed, I think you can remove the map line entirely
* instead of filter, use any with both conditions with &&


Final code would be:
`oldHistory?.let { old ->
            old.filterIsInstance<LinkedChatItem>().any { oldMessage ->
                oldMessage.messageId != null && oldMessage.messageId == newMessage.id
            }
        } ?: true`

Reply:
You lost to call `.not()`. But yeah, this can be simplified. But I prefer not to use `?.let` in the beggining.